service/src/main/java/io/camunda/service/ApiServices.java

```diff
@@ -55,7 +55,9 @@ public T withAuthentication(
   }
 
   protected <R> CompletableFuture<R> sendBrokerRequest(final BrokerRequest<R> brokerRequest) {
-    brokerRequest.setAuthorization(authentication.token());
+    if (authentication != null) {
```

Comment:
❓ We should validate this change with someone more familiar with the wider code than me, it seems fine but I wonder if maybe @houssain-barouni could see something here that may cause an issue?

Reply:
It seems this change was necessary because the broker calls to `UserServices.search` and `UserServices.createUser` within `DataInitializer` without passing the authentication.

When the gateway operates in standalone mode and needs to communicate with a remote broker, authorization may be needed. You should use `userServices.withAuthorization(RequestMapper.getAuthentication()).createUser(..)` for these calls.

Since data initialization occurs outside the scope of an HTTP request, `RequestMapper.getAuthentication()` might fail to retrieve the authorized tenants list (`List<String> authorizedTenants = TenantAttributeHolder.tenantIds()`), which needs to be confirmed. If this is the case, you may need to create a new method that generates the authentication object without tenants, as they might not be necessary for this use case.

Reply:
Thanks @houssain-barouni for clarification, what I face here is that it's the calls to create a user and before that we don't have any user available to be able to pass the basic auth. Do you know any workaround in this case?